woocommerce/woocommerce
# Add support for defining WP, WC & PHP version for e2e & api tests

### Description

This PR adds support for defining specific WordPress, WooCommerce & PHP versions for the `wp-env` environment.

### Changes proposed in this Pull Request:

- Adds a `UPDATE_WP_JSON` class which will update the respective fields in the `.wp-env.json` file.
   - This can be leveraged by first setting the `UPDATE_WP_JSON_FILE=1` env var along with any or all of the following vars:
      - `WP_VERSION`
        - Acceptable versions are `nightly`, `trunk`, and any version listed on [WordPress Releases](https://github.com/WordPress/WordPress/tags)
      - `WC_VERSION`
      - `PHP_VERSION`
- Normalizes the contents of the `wp-content` directory in the container to ensure that we don't run into permission errors when defining a specific version of WordPress.
- Uses `docker-compose` directly to avoid permission errors.
- Merges the multiple commands in `test-env-setup.sh` to reduce setup time.

### Opportunity 
These changes would allow us the ability to setup actions matrix that will allow us to target a variety of WP, WC & PHP Versions.

### How to test the changes in this Pull Request:

1. Checkout this branch 
2. Navigate to the plugin directory with `cd plugins/woocommerce`
3. Build the container with specific version by running `UPDATE_WP_JSON_FILE=1 WP_VERSION=6.1.1 WC_TEST_VERSION=7.4.1 PHP_VERSION=8.2 pnpm run env:test`
4. Confirm that the `.wp-env.json` file is updated accordingly.
5. Destroy the environment and run step 3 with of the version var removed.
6. Confirm that the `.wp-env.json` file is updated accordingly.
7. Run E2E/API tests with `pnpm run test:e2e-pw` / `pnpm run test:api-pw`
8. Destroy the environment and revert the `.wp-env.json` changes by running `pnpm run env:destroy`
9. Confirm that all the changes to `.wp-env.json` have been reverted.
